### ISSUE TITLE
Can't able to set request header to Access-Control-Allow-Origin🤗 

### ISSUE THREAD

**Issue by sidmohanty11** [gofiber/fiber#1441]
**Question description**
Guys I am a big fan of your work, I was just making an api to work with my react app. But Access-Control-Allow-Origin cors issue  is coming up. I checked AllowOrigins was set to * in default, I manually tried it too. Then I tried to add to AllowHeaders but nothing seem to work. I read upon the Config struct and it seem to set it when its either "*" or a specific domain but in my case none works.  I am doing something wrong? 

**Code snippet** _Optional_

```go
	app.Use(cors.New(cors.Config{
		AllowCredentials: true,
		AllowOrigins:     "https://gradbook-cet.netlify.com",
		AllowHeaders:     "Origin, Content-Type, Accept, Accept-Language, Content-Length",
	}))

```

**Comment by ReneWerner87** [comment]
hi, can you provide a code example that contains a minimal setup and includes the problem.

only with the code call i can't tell you what you are doing wrong.

**Comment by sidmohanty11** [comment]
github for frontend -> https://github.com/sidmohanty11/gradbook-client
github for backend -> https://github.com/sidmohanty11/gradbook-api

@ReneWerner87 

my main.go file,
```go
func main() {
	// server connection
	app := fiber.New()

	PORT, err := getPort()

	if err != nil {
		log.Fatal(err)
	}

	app.Use(cors.New(cors.Config{
		AllowCredentials: true,
		AllowOrigins:     "https://gradbook-cet.netlify.com",
		AllowHeaders:     "Origin, Content-Type, Accept, Accept-Language, Content-Length",
	}))

	app.Use(logger.New())

	// db connection
	db, err := db.ConnectSQL()

	if err != nil {
		log.Fatalln("Cannot connect to DB!")
	}

	fmt.Println("Connected to DB at PORT 5432")
	defer db.SQL.Close()

	routes.Setup(app)
	handlers.NewRepo(db)
	// server listening port
	app.Listen(PORT)

	fmt.Printf("Listening at PORT%s", PORT)
}
```

frontend call,
```javascript
const res = await axios
      .post(
        "/api/v1/login",
        {
          username,
          password,
        },
        { withCredentials: true }
      )
      .catch((err) => setAuthorized(false));
```
my problem,
![error-cors](https://user-images.githubusercontent.com/73601258/126131216-5b684065-dd89-45c1-94f5-dd68293f21ca.png)

**Comment by ReneWerner87** [comment]
But
![image](https://user-images.githubusercontent.com/7063188/126658846-e40746df-bf05-48e4-b37f-82e7e0f774f0.png)
is different with
![image](https://user-images.githubusercontent.com/7063188/126658900-6d2a293c-f34b-4999-a34e-c508bcffc098.png)


"****.app" vs "****.com"
@sidmohanty11

**Comment by sidmohanty11** [comment]
yeah, that's so embarrassing, @ReneWerner87  so sorry
![herku1](https://user-images.githubusercontent.com/73601258/126666763-ac117f00-3634-44e8-a7c4-6c9404afe7de.png)
![herku2](https://user-images.githubusercontent.com/73601258/126666774-f6882bf8-24b0-4383-a8b9-689e46046c71.png)
![herku3](https://user-images.githubusercontent.com/73601258/126666782-5a4c1f1a-c65c-4eba-b137-8d56312c7192.png)
Do you think this (Access-Control-Allow-Origin) header should be explicitly mentioned because it's not showing? But the way I have set up routes is like` /api/v1/...`,and the origin is set to `https://gradbook-cet.netlify.app/` but still its showing origin as https://gradbook-cet.netlify.app and referer as` https://gradbook-cet.netlify.app/` .Do you think that can be the problem?? Thank you so much for looking into this

**Comment by ReneWerner87** [comment]
Hmm, not sure what you mean right now

The origin header is sent automatically by the browser and should match the one in the cors middleware that you want to allow access to

**Comment by ReneWerner87** [comment]
If you are satisfied with a lower security level, you can also change it to the wildcard.

**Comment by ReneWerner87** [comment]
https://owasp.org/www-community/attacks/CORS_OriginHeaderScrutiny

**Comment by sidmohanty11** [comment]
![herku4_LI](https://user-images.githubusercontent.com/73601258/126673094-0610135e-0280-4de9-bff4-630eab41118d.jpg)
@ReneWerner87 how can I set this then? Its value is null I guess.

**Comment by ReneWerner87** [comment]
Check this https://github.com/gofiber/fiber/issues/1411#issuecomment-869518111

**Comment by sidmohanty11** [comment]
tried to use the headers you mentioned and used wildcard * for origin, still can't get it to work

**Comment by panyox** [comment]
noticed that there is a 404 error before CORS, so i think maybe you should check out the url path is correct?

**Comment by sidmohanty11** [comment]
@panyox  sorry for the late reply, that 404 is due to favicon. Do you think, maybe that can be the problem?

**Comment by imsk17** [comment]
You have to allow the "OPTIONS" method as well. You can do so by passing a string with your desired methods to the `AllowMethods` param. i.e.

`
app.Use(cors.New(
		cors.Config{
			AllowOrigins: "..."
			AllowMethods: "GET, POST, HEAD, PUT, DELETE, PATCH, OPTIONS",
		},
	))
`

**Comment by ReneWerner87** [comment]
![image](https://user-images.githubusercontent.com/7063188/127825138-b4b199ef-b18e-41e2-8ed6-c36a602577ac.png)
can you share your config again, here is another host, just use the wildcard for the domains in the config

> how can I set this then? Its value is null I guess.
you cannot set this, it is set automatically by our middleware

**Comment by ReneWerner87** [comment]
https://github.com/gofiber/fiber/blob/master/middleware/cors/cors.go#L148

**Comment by ReneWerner87** [comment]
maybe more headers are sent than you(@sidmohanty11 ) allow

you should try it from my point of view, first with the default config, here everything is allowed
```go
app.Use(cors.New())
```

**Comment by sidmohanty11** [comment]
@ReneWerner87 Thank you so much, it worked.

**Comment by WilliamSilveiraF** [comment]
@sidmohanty11 Where did you deploy the backend application? I have the same error

https://github.com/sidmohanty11/gradbook-api/

**Comment by sidmohanty11** [comment]
@WilliamSilveiraF I deployed it on heroku, try using wildcard or the default config

**Comment by jub0bs** [comment]
@sidmohanty11 About https://github.com/gofiber/fiber/issues/1441#issuecomment-885013511: a Web origin does not contain a path. Dropping that trailing slash should solve your issue.

**Comment by yacinebenkaidali** [comment]
It worked after adding 
```go
context.Set("Access-Control-Allow-Origin", c.GetReqHeaders()["Origin"])
```

**Comment by jub0bs** [comment]
@yacinebenkaidali

Either you want to allow only anonymous requests from any Web origin, in which case you can use `Access-Control-Allow-Origin: *`; or you want to allow credentialed requests only from Web origins in an allowlist.

But you should never unconditionally reflect the value of the `Origin` request header in the `Access-Control-Allow-Origin` response header, especially if you also allow credentials. Doing so is very insecure!

**Comment by yacinebenkaidali** [comment]
@jub0bs Thank you so much for the information, i didn't knew that before

**Comment by wade-liwei** [comment]
@yacinebenkaidali
Could you please share me your full codes, I do not know how to add the code line.

context.Set("Access-Control-Allow-Origin", c.GetReqHeaders()["Origin"])

**Comment by wade-liwei** [comment]
I tried the default cors configure, but it does not work.

	"github.com/gofiber/fiber/v2/middleware/cors"	
	app.Use(cors.New())
	
Access to fetch at 'https://api.dev.a232343242.ai/api/v1/predictions' from origin 'http://192.168.1.220:8080' has been blocked by CORS policy: Request header field apikey is not allowed by Access-Control-Allow-Headers in preflight response.

**Comment by jub0bs** [comment]
@wade-liwei Read the error message carefully. It's telling you that you need to list `"apikey"` in your CORS configuration's `AllowHeaders` field. See https://docs.gofiber.io/api/middleware/cors#custom-configuration-specific-origins-headers-etc

**Comment by wade-liwei** [comment]
yea, but the problem is that I can not list "apikey" in my CORS configuration's AllowHeaders field.

still does not work.

	app.Use(cors.New())

	app.Use(cors.New(cors.Config{
		AllowOrigins: "https://gofiber.io, https://gofiber.net,https://api.dev.aiai123423.ai",
		AllowHeaders: "Origin, Content-Type, Accept,apikey,Apikey",
	}))

**Comment by jub0bs** [comment]
@wade-liwei  Are you using both calls to `app.Use`? You should configure CORS at most once.

Also, you don't need to list `Origin` in `AllowHeaders` and you don't need to specify `ApiKey` in various cases, since headers are case-insensitive.

Finally, you're not allowing the necessary Web origins in your CORS config. Again, read the CORS error message carefully:

```
Access to fetch at 'https://api.dev.a232343242.ai/api/v1/predictions' from origin 'http://192.168.1.220:8080' has been blocked by CORS policy [...]
```

You need to list (at least) `http://192.168.1.220:8080` in the `AllowOrigins` field.

**Comment by wade-liwei** [comment]
1. only one call.
2. I tried the following configure,but still does not work. 	
app.Use(cors.New(cors.Config{AllowOrigins: "*", AllowHeaders: "Origin, Content-Type, Accept, Accept-Language, Content-Length, Apikey, apikey", AllowMethods: "GET, POST, HEAD, PUT, DELETE, PATCH, OPTIONS"}))

3. I tried list http://192.168.1.220:8080 in the AllowOrigins field.still does not work. 
	app.Use(cors.New(cors.Config{
		AllowOrigins: "http://192.168.1.220:8080",
	}))

always get the errors:

Access to fetch at 'https://api.dev.simulate123423.ai/api/v1/predictions' from origin 'http://192.168.1.220:8080' has been blocked by CORS policy: Request header field apikey is not allowed by Access-Control-Allow-Headers in preflight response.


Could you please give me an example code.thanks.

**Comment by jub0bs** [comment]
@wade-liwei You'd be better off asking on Stack Overflow (where I'm also active) than in a closed GitHub issue.

**Comment by gaby** [comment]
@wade-liwei Please create a new issue or join the Discord.

**Comment by wade-liwei** [comment]
got it, thanks.

**Comment by drnocam** [comment]
If you use reverse proxy , you cannot set gofiber cors policy.
